File: LeetCode/python3/985.py
class Solution:
    def sumEvenAfterQueries(self, A: 'List[int]', queries: 'List[List[int]]') -> 'List[int]':
        m = sum(a for a in A if a % 2 == 0)
        ret = []
        for q in queries:
            m -= A[q[1]] if A[q[1]] % 2 == 0 else 0
            A[q[1]] += q[0]
            m += A[q[1]] if A[q[1]] % 2 == 0 else 0
            ret.append(m)
        return ret

File: LeetCode/cpp/985.cpp

class Solution {
public:
    vector<int> sumEvenAfterQueries(vector<int> &A, vector<vector<int>> &queries) {
        vector<int> ret;
        int sum = 0;
        for (auto &a : A) {
            if (a % 2 == 0)
                sum += a;
        }
        for (int i = 0; i < queries.size(); ++i) {
            int &temp = A[queries[i][1]];
            if (temp % 2 == 0)
                sum -= temp;
            temp += queries[i][0];
            if (temp % 2 == 0)
                sum += temp;
            ret.push_back(sum);
        }
        return ret;
    }
};